### PR TITLE
Soback 81 add spinner to facets

### PR DESCRIPTION
**Pull request by jesperlauridsen**
Added spinner to facets
Added variable for keeping track of facet loading status
Moved spinner from harvestTimes to global (this is done in another PR too, so this might cause some merge conflicts when we get there)
Fixed a bug where pagnation also resulted in a facet request - this has been taken care of now

The spinner can be tested by adding setTimeout to the response from the backend.

### CODE REVIEW

File: src/js/src/components/SearchFacetOptions.vue [hunk 1]
```diff
@@ -1,15 +1,18 @@
...
-    <div v-for="(facetCategory, index) in Object.entries(facets.facet_fields)" :key="index" class="facetCategory">
-      <div class="facetCategoryName">
-        {{ facetCategory[0] }}
-      </div> 
-      <div v-for="(facet, facetIndex) in facetCategory[1]"
-           :key="facetIndex"
-           :class="facetIndex % 2 === 0 ? 'facetItem' : 'facetCount'"
-           @click="facetIndex % 2 === 0 ? applyFacet(facetCategory[0], facet) : null">
-        {{ facetIndex % 2 === 0 ? facet || "Unknown" : "(" + facet + ")" }}
+    <div v-if="facetLoading && !loading" class="spinner" />
```
Comment (jorntx) [inline src/js/src/components/SearchFacetOptions.vue:4]: Ahh clever - wait for main spinner to finish.

File: src/js/src/views/SolrWayback.vue [hunk 2]
```diff
@@ -74,18 +74,25 @@ export default {
...
        to.query.grouping !== from.query.grouping ||
        to.query.facets !== from.query.facets)
+    },
+    checkIfRoutingIsPageTurn(to, from) {
+      return to.query.query === from.query.query &&
+                  to.query.imgSearch === from.query.imgSearch && 
+                  to.query.urlSearch === from.query.urlSearch && 
+                  to.query.grouping === from.query.grouping &&
+                  to.query.facets === from.query.facets &&
+                  to.query.offset !== from.query.offset
```
Comment (jorntx) [inline src/js/src/views/SolrWayback.vue:84]: I feel there is a smarter way to make this comparison but haven't figured it out yet:)

**Review by jorntx — APPROVED** [review]
Code looks good - only a minor comment.

Very hard to test the spinner locally because response time is blazingly fast but nothing breaks.

**Comment by jorntx** [comment]
@jesperlauridsen same with the conflicts here - thanks 🙏

**Comment by jesperlauridsen** [comment]
Fixed merge conflicts.

**Pull request merged**